framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java

```diff
@@ -23,10 +23,9 @@
     public static void annotateSupers(
             List<AnnotatedTypeMirror.AnnotatedDeclaredType> supertypes,
             TypeElement subtypeElement) {
-
-        final boolean isInterface = subtypeElement.getSuperclass().getKind() == TypeKind.NONE;
-
-        final int typeOffset = isInterface ? 0 : -1;
+        // typeOffset should always be -1. For a subtypeElement which represents an interface
```

Comment:
Try to explain the point of typeOffset without relying on knowing about the previous implementation. That is, simply explain why -1 is the right offset.
Maybe it's enough to rewrite line 32 as `final int typeIndex = i - 1;` and explain the -1 there.